core/dbt/adapters/base/impl.py

```diff
@@ -907,13 +910,17 @@ def convert_time_type(cls, agate_table: agate.Table, col_idx: int) -> str:
 
     @available
     @classmethod
-    def convert_type(cls, agate_table, col_idx):
+    def convert_type(
+        cls, agate_table: agate.Table, col_idx: int
+    ) -> Optional[str]:
         return cls.convert_agate_type(agate_table, col_idx)
 
     @classmethod
-    def convert_agate_type(cls, agate_table, col_idx):
-        agate_type = agate_table.column_types[col_idx]
-        conversions = [
+    def convert_agate_type(
+        cls, agate_table: agate.Table, col_idx: int
+    ) -> Optional[str]:
```

Comment:
This is what the old code actually did, though I'm not confident it was the intent... maybe it should raise if no match is found and then it can return `str` (and so can `convert_type`).

Reply:
I think returning `str` instead of `None` here is a good idea, but I don't know that we need/want to make that functional change in this PR

Reply:
Yeah, that was my thought - I didn't want to let this PR get any bigger.